tools/generate-update-graph/main.go

```diff
@@ -0,0 +1,233 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/google/go-github/v43/github"
+	"sigs.k8s.io/yaml"
+
+	"github.com/authzed/spicedb-operator/pkg/config"
+	"github.com/authzed/spicedb-operator/pkg/updates"
+)
+
+//go:generate go run main.go ../../default-operator-config.yaml
+
+const (
+	githubNamespace  = "authzed"
+	githubRepository = "spicedb"
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		fmt.Println("must provide filename")
+		os.Exit(1)
+	}
+	ctx := context.Background()
+	client := github.NewClient(nil)
+
+	// this returns the newest release by date, not by version
+	// note that spicedb uses the same API to determine if it's up to date
+	latestRelease, _, err := client.Repositories.GetLatestRelease(ctx, githubNamespace, githubRepository)
+	if err != nil {
+		panic(err)
+	}
+
+	opconfig := config.OperatorConfig{
+		ImageName: "ghcr.io/authzed/spicedb",
+		UpdateGraph: updates.UpdateGraph{Channels: []updates.Channel{
+			postgresChannel(),
+			crdbChannel(),
+			mysqlChannel(),
+			spannerChannel(),
+		}},
+	}
+
+	for _, c := range opconfig.Channels {
+		if c.Nodes[0].Tag != *latestRelease.Name {
+			fmt.Printf("channel %q does not contain the latest release %q\n", c.Name, *latestRelease.Name)
+			os.Exit(1)
+			return
+		}
+	}
+
+	yamlBytes, err := yaml.Marshal(&opconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	if err := os.WriteFile(os.Args[1], yamlBytes, 0o666); err != nil {
+		panic(err)
+	}
+}
+
+func postgresChannel() updates.Channel {
```

Comment:
there's some low hanging fruit to make generating these lists a little better. I was holding off because any helpers I added here would require a ton of testing, and because some other decisions might affect how we do it (i.e. if we commit to semver, that gives an easy way to talk about ranges of versions. if we don't, we'll need to track indexes in the ordered list of nodes).